File: tiq-e/strings/7/main.ts
import { performance } from "perf_hooks";

function strStr(haystack: string, needle: string): number {
  if (needle === "") return 0;
  let l = needle.length;

  for (let i = 0; i < haystack.length - needle.length + 1; i++) {
    if (haystack.slice(i, i + l) == needle) return i;
  }
  return -1;
}

const haystack = "hello";
const needle = "ll";

const start = performance.now();
const result = strStr(haystack, needle);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/strings/7/main.go
package main

import (
	"fmt"
	"time"
)

func strStr(haystack string, needle string) int {
	if needle == "" {
		return 0
	}
	length := len(needle)
	for i := 0; i < len(haystack)-len(needle)+1; i++ {
		if haystack[i:i+length] == needle {
			return i
		}
	}
	return -1
}

func main() {
	var haystack = "hello"
	var needle = "ll"

	var start = time.Now()
	var result = strStr(haystack, needle)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
